RoktDemo/ViewModels/CustomerDetailsViewModel.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  CustomerDetailsViewModel.swift
+//  RoktDemo
+//
+//  Licensed under the Rokt Software Development Kit (SDK) Terms of Use
+//  Version 2.0 (the "License");
+//
+//  You may not use this file except in compliance with the License.
+//
+//  You may obtain a copy of the License at https://rokt.com/sdk-license-2-0/
+
+import Foundation
+class CustomerDetailsViewModel: ObservableObject {
+    let customerDetails: CustomerDetailsModel
+    let accountDetail: AccountDetailViewData
+    
+    @Published var state: String = ""
+    @Published var postcode: String = ""
+    @Published var countries: [String] = []
+    @Published var selectedCountry: String = ""
+    @Published var advancedDetailsKV: [KeyValue] = []
+    
+    internal init(customerDetails: CustomerDetailsModel,
+                  advancedDetails: [String: String],
+                  accountDetail: AccountDetailViewData) {
+        self.customerDetails = customerDetails
+        self.accountDetail = accountDetail
+        
+        state = customerDetails.state
+        postcode = customerDetails.postcode
+        countries = customerDetails.country
+        selectedCountry = countries.first ?? ""
+        
+        advancedDetailsKV = []
+        for (key, value) in advancedDetails {
+            advancedDetailsKV.append(KeyValue(key: key, value: value))
+        }
+        // add empty key value at the end
```

Comment:
More comments please. I dont know why we need an empy key value pair.

Reply:
Done